### ISSUE TITLE
aspell segmentation fault on building dictionary

### ISSUE THREAD

**Issue by ncihnegn** [Alexpux/MINGW-packages#904]
Shall I downgrade winpthreads?

==> Starting build()...
Finding Dictionary file location ... /mingw64/lib/aspell-0.60
Finding Data file location ... /mingw64/lib/aspell-0.60
/mingw64/bin/prezip-bin -d < en-common.cwl | /mingw64/bin/aspell  --lang=en create master ./en-common.rws
/mingw64/bin/prezip-bin -d < en-variant_0.cwl | /mingw64/bin/aspell  --lang=en create master ./en-variant_0.rws
/mingw64/bin/prezip-bin -d < en-variant_1.cwl | /mingw64/bin/aspell  --lang=en create master ./en-variant_1.rws
/mingw64/bin/prezip-bin -d < en-variant_2.cwl | /mingw64/bin/aspell  --lang=en create master ./en-variant_2.rws
/mingw64/bin/prezip-bin -d < en-w_accents-only.cwl | /mingw64/bin/aspell  --lang=en create master ./en-w_accents-only.rws
/mingw64/bin/prezip-bin -d < en-wo_accents-only.cwl | /mingw64/bin/aspell  --lang=en create master ./en-wo_accents-only.rws
/mingw64/bin/prezip-bin -d < en_CA-variant_0.cwl | /mingw64/bin/aspell  --lang=en create master ./en_CA-variant_0.rws
/mingw64/bin/prezip-bin -d < en_CA-variant_1.cwl | /mingw64/bin/aspell  --lang=en create master ./en_CA-variant_1.rws
/mingw64/bin/prezip-bin -d < en_CA-w_accents-only.cwl | /mingw64/bin/aspell  --lang=en create master ./en_CA-w_accents-only.rws
/mingw64/bin/prezip-bin -d < en_CA-wo_accents-only.cwl | /mingw64/bin/aspell  --lang=en create master ./en_CA-wo_accents-only.rws
/bin/sh: line 1:  1660 Done                    /mingw64/bin/prezip-bin -d < en-wo_accents-only.cwl
      7388 Segmentation fault      | /mingw64/bin/aspell --lang=en create master ./en-wo_accents-only.rws
Makefile:108: recipe for target 'en-wo_accents-only.rws' failed
make: **\* [en-wo_accents-only.rws] Error 139
make: **\* Waiting for unfinished jobs....
/bin/sh: line 1:  6500 Done                    /mingw64/bin/prezip-bin -d < en_CA-w_accents-only.cwl
      8940 Segmentation fault      | /mingw64/bin/aspell --lang=en create master ./en_CA-w_accents-only.rws
Makefile:108: recipe for target 'en_CA-w_accents-only.rws' failed
make: **\* [en_CA-w_accents-only.rws] Error 139
/bin/sh: line 1:  6960 Done                    /mingw64/bin/prezip-bin -d < en_CA-wo_accents-only.cwl
      8424 Segmentation fault      | /mingw64/bin/aspell --lang=en create master ./en_CA-wo_accents-only.rws
Makefile:108: recipe for target 'en_CA-wo_accents-only.rws' failed
make: **\* [en_CA-wo_accents-only.rws] Error 139

**Comment by ncihnegn** [comment]
Verified: downgraded winpthreads works.

**Comment by DavidEGrayson** [comment]
You're not the only one having trouble with threading.  There was another recent report about threading causing segmentation faults here: https://github.com/Alexpux/MINGW-packages/pull/905

**Comment by DavidEGrayson** [comment]
What package are you trying to build?  I just built mingw-w64-aspell on my machine after updating all my MSYS2 packages.

Here is a snippet of my build output, which looks very different from yours, which makes me wonder what you are building:

~~~~
==> Starting build()...
checking build system type... x86_64-w64-mingw32
checking host system type... x86_64-w64-mingw32
checking target system type... x86_64-w64-mingw32
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
~~~~

**Comment by Ede123** [comment]
Probably the same issue described at https://sourceforge.net/p/msys2/tickets/180/

Is there a known fix that does not include downgrading winpthreads? Would certainly make life much easier, especially it would be futureproof...

**Comment by Ede123** [comment]
Even more possible instances of this bug:
* https://sourceforge.net/p/mingw-w64/bugs/432/
* https://bugs.launchpad.net/inkscape/+bug/1412365
* https://bugs.launchpad.net/inkscape/+bug/1714278

Unfortunately temporarily downgrading winpthreads is not an option if a program is linked against Aspell (like Inkscape).

**Comment by Ede123** [comment]
Just created a PR which allows to build Aspell with gcc 7 in case somebody wants to try to build / debug this...

**Comment by Ede123** [comment]
Fixed with the patch from @jtanx in #2872.

Patch also submitted upstream as https://github.com/GNUAspell/aspell/pull/532
